### ISSUE TITLE
Need Importer for Facet, Affectiva, and OpenFace

### ISSUE THREAD

**Issue by ljchang** [cosanlab/feat#2]
Importer should make sure that files are correctly imported as a Fex object.  `feat.Fex._check_if_fex` function is a starting place.

**Comment by jcheong0428** [comment]
Need to think about how to standardize naming of features. 


**FACET:**
Emotions:
Joy,Anger,Surprise,Fear,Contempt,Disgust,Sadness,Confusion,Frustration,Neutral,Positive,Negative
Action Units
AU1, AU2,AU4,AU5,AU6,AU7,AU9,AU10,AU12,AU14,AU15,AU17,AU18,AU20,AU23,AU24,AU25,
 AU26, AU28, AU43, 
Misc
Yaw Degrees,Pitch Degrees,Roll Degrees

**Affectiva:**
Also see here : https://developer.affectiva.com/metrics/
'anger', 'attention', 'browFurrow', 'browRaise', 'cheekRaise',
       'chinRaise', 'contempt', 'dimpler', 'disgust', 'engagement',
       'eyeClosure', 'eyeWiden', 'fear', 'innerBrowRaise', 'jawDrop', 'joy',
       'lidTighten', 'lipCornerDepressor', 'lipPress', 'lipPucker',
       'lipStretch', 'lipSuck', 'mouthOpen', 'noseWrinkle', 'sadness', 'smile',
       'smirk', 'surprise', 'upperLipRaise', 'valence

**OpenFace:** 
AU1, 2, 4, 5, 6, 7, 8, 9, 10, 12, 14, 15, 17, 20, 23, 25, 26, 45
Presence and Intensity for each AU provided.

**Comment by ljchang** [comment]
I think subclassing the base Fex class is the way to go.  If we have an importer function it can auto detect the file type and figure out the correct class.

I initially made it possible to load text files similar to Brain_Data and Adjacency, but was having issues with the DataFrame constructor.  Probably makes more sense to just have an importer function like pandas (e.g., pd.read_csv())

**Comment by ljchang** [comment]
I saw this on another pandas spinoff and thought we might be able to use something like it.

```
@classmethod
def from_file(cls, filename, **kwargs):
    """Alternate constructor to create a ``GeoDataFrame`` from a file.
    Can load a ``GeoDataFrame`` from a file in any format recognized by
    `fiona`. See http://toblerity.org/fiona/manual.html for details.
    Parameters
    ----------
    filename : str
        File path or file handle to read from. Depending on which kwargs
        are included, the content of filename may vary. See
        http://toblerity.org/fiona/README.html#usage for usage details.
    kwargs : key-word arguments
        These arguments are passed to fiona.open, and can be used to
        access multi-layer data, data stored within archives (zip files),
        etc.
    Examples
    --------
    >>> df = geopandas.GeoDataFrame.from_file('nybb.shp')
    """
    return geopandas.io.file.read_file(filename, **kwargs)
```

**Comment by jcheong0428** [comment]
I'm keen on trying the abstract method described here. 
That way we can just call feat.read_file() and it will figure out the subclass. 

https://julien.danjou.info/blog/2013/guide-python-static-class-abstract-methods

**Comment by Nathaniel-Haines** [comment]
OpenFace files contain many features. Are we interested in just the AU-related features or should we include the facial landmarks too? See here for all relevant feature info: https://github.com/TadasBaltrusaitis/OpenFace/wiki/Output-Format#featureextraction

**Comment by ljchang** [comment]
I think we should grab all of them for now.  That's easy to change later, but I suspect we might want to use some of them as confounds in the `.clean()` method.

**Comment by Nathaniel-Haines** [comment]
Sounds good. I am working on the OpenFace file importer now. FYI—openface output files have a white space before most of the column names (e.g. ' AU01_r'). I assume we should remove these upon importing the files?

**Comment by ljchang** [comment]
Let's definitely remove the white space.

**Comment by ljchang** [comment]
@jcheong0428: don't forget to add a basic example of how to use this in the dropbox papers tutorials.  Going to close this issue for now.  Fixed in PR #16
